From 59a1d7d6b2f6677e6f4ff2186a1257fb7a443431 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Wed, 2 Sep 2015 15:47:32 +0900
Subject: [PATCH 85/85] arm64: dts: r8a7795: Add GSX node

Add GSX device node for r8a7795, include clocks, and interrupt
properties.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a7795.dtsi  |   28 ++++++++++++++++++++++++++++
 include/dt-bindings/clock/r8a7795-clock.h |    1 +
 2 files changed, 29 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index 3d1689f..802161a 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -182,6 +182,15 @@
 				clock-output-names = "sd3";
 			};
 
+			zg_clk: zg_clk {
+				compatible = "fixed-factor-clock";
+				clocks = <&cpg_clocks R8A7795_CLK_PLL4>;
+				#clock-cells = <0>;
+				clock-div = <2>;
+				clock-mult = <1>;
+				clock-output-names = "zg";
+			};
+
 			cpg_clocks: cpg_clocks@e6150000 {
 				#address-cells = <2>;
 				#size-cells = <2>;
@@ -195,6 +204,18 @@
 				clock-output-names = "main", "pll0", "pll1","pll2",
 						     "pll3", "pll4";
 
+				mstp1_clks: mstp1_clks@e6150134 {
+					compatible = "renesas,r8a7795-mstp-clocks",
+						     "renesas,cpg-mstp-clocks";
+					reg = <0 0xe6150134 0 4>, <0 0xe6150038 0 4>;
+					clocks = <&zg_clk>;
+					#clock-cells = <1>;
+					renesas,clock-indices = <
+								 R8A7795_CLK_3DGE
+					>;
+					clock-output-names = "3dge";
+				};
+
 				mstp2_clks: mstp2_clks@e6150138 {
 					compatible = "renesas,r8a7795-mstp-clocks",
 						     "renesas,cpg-mstp-clocks";
@@ -776,5 +797,12 @@
 			phy-names = "usb";
 			status = "disabled";
 		};
+
+		gsx: gsx@fd000000 {
+			compatible = "renesas,gsx";
+			reg = <0 0xfd000000 0 0x3ffff>;
+			interrupts = <GIC_SPI 119 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7795_CLK_3DGE>;
+		};
 	};
 };
diff --git a/include/dt-bindings/clock/r8a7795-clock.h b/include/dt-bindings/clock/r8a7795-clock.h
index e1d4718..ae1980b 100644
--- a/include/dt-bindings/clock/r8a7795-clock.h
+++ b/include/dt-bindings/clock/r8a7795-clock.h
@@ -20,6 +20,7 @@
 /* MSTP0 */
 
 /* MSTP1 */
+#define R8A7795_CLK_3DGE		12
 
 /* MSTP2 */
 #define R8A7795_CLK_SCIF0		7
-- 
1.7.9.5

